src/test/resources/example-data/quoted_fields_with_separator.csv

```diff
@@ -0,0 +1,5 @@
+header_1,header_2,"weird, but valid header 3", header_4_with_extra_whitespace
+test,2,3,4
+"test,test,test,test",3,3,5
+" even weirder, but still valid, value with extra whitespaces that remain due to quoting /  ",1,2,3
```

Comment:
Would it also work with newlines in quoted values?

Reply:
It actually would ... I'd consider that to be moving into abomination territory, but it does work :)
Added tests for this in https://github.com/snowlift/trino-storage/pull/244/commits/984316f2fa82431bb264a801fd67131169084f04